File: output/deepseek_chat/codenet/direct_translation/python/java/s016605598.java

import java.util.*;

public class s016605598 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int const_ = 0;
        PriorityQueue<Integer> left = new PriorityQueue<>(Collections.reverseOrder());
        PriorityQueue<Integer> right = new PriorityQueue<>();
        int q = sc.nextInt();
        int cnt = 0;
        int leftSum = 0;
        int rightSum = 0;
        for (int i = 0; i < q; i++) {
            int[] t = Arrays.stream(sc.nextLine().split(" ")).mapToInt(Integer::parseInt).toArray();
            if (t.length == 1) {
                t = Arrays.stream(sc.nextLine().split(" ")).mapToInt(Integer::parseInt).toArray();
            }
            if (t[0] == 1) {
                int a = t[1];
                int b = t[2];
                const_ += b;
                if (cnt % 2 == 0) {
                    if (right.isEmpty()) {
                        left.add(a);
                        leftSum += a;
                        cnt++;
                        continue;
                    }
                    int c = right.poll();
                    rightSum -= c;
                    if (a <= c) {
                        left.add(a);
                        right.add(c);
                        leftSum += a;
                        rightSum += c;
                    } else {
                        left.add(c);
                        right.add(a);
                        leftSum += c;
                        rightSum += a;
                    }
                } else {
                    int c = left.poll();
                    leftSum += c;
                    if (a <= -c) {
                        left.add(-a);
                        right.add(-c);
                        leftSum += a;
                        rightSum -= c;
                    } else {
                        left.add(c);
                        right.add(a);
                        leftSum -= c;
                        rightSum += a;
                    }
                }
                cnt++;
            } else {
                int tmp = -left.peek();
                System.out.println(tmp + " " + (const_ - leftSum + rightSum + tmp * (cnt % 2)));
            }
        }
    }
}

File: dataset/codenet/Python/Code/s016605598.py
import heapq
const = 0
left = []
right = []

q = int(input())
cnt = 0
left_sum = 0
right_sum = 0
for _ in range(q):
    t = list(map(int, input().split()))
    if t[0] == 1:
        # update
        a, b = t[1:]
        const += b
        if cnt % 2 == 0:
            if not right:
                heapq.heappush(left, -a)
                left_sum += a
                cnt += 1
                #print(left_sum, right_sum, cnt)
                continue

            c = heapq.heappop(right)
            right_sum -= c
            if a <= c:
                heapq.heappush(left, -a)
                heapq.heappush(right, c)
                left_sum += a
                right_sum += c
            else:
                heapq.heappush(left, -c)
                heapq.heappush(right, a)
                left_sum += c
                right_sum += a
        else:
            c = heapq.heappop(left)
            left_sum += c
            if a <= -c:
                heapq.heappush(left, -a)
                heapq.heappush(right, -c)
                left_sum += a
                right_sum -= c
            else:
                heapq.heappush(left, c)
                heapq.heappush(right, a)
                left_sum -= c
                right_sum += a
        cnt += 1
        #print(left_sum, right_sum, cnt)
    else:
        tmp = -left[0]
        print(tmp, const - left_sum + right_sum + tmp * (cnt % 2))

